packages/kotti-input-number/src/InputNumber.vue

```diff
@@ -72,29 +69,66 @@ export default {
 		increaseButtonStyle() {
 			return {
 				'kt-input-number__button': true,
-				'kt-input-number__button--disabled':
-					this.currentValue == this.max || this.disabled,
+				'kt-input-number__button--disabled': this.incrementDisabled,
 			}
 		},
 		decreaseButtonStyle() {
 			return {
 				'kt-input-number__button': true,
-				'kt-input-number__button--disabled':
-					this.currentValue == this.min || this.disabled,
+				'kt-input-number__button--disabled': this.decrementDisabled,
 			}
 		},
 	},
 	watch: {
 		value(val) {
 			this.currentValue = parseFloat(val)
 		},
+		currentValue(newVal) {
+			if (!this.formError) {
+				this.$emit('input', newVal)
+			}
+			if (typeof this.min === 'number') {
+				if (newVal - this.step < this.min) {
+					this.decrementDisabled = true
+				} else if (!this.disabled) {
+					this.decrementDisabled = false
+				}
+			}
+			if (typeof this.max === 'number') {
+				if (newVal + this.step > this.max) {
+					this.incrementDisabled = true
+				} else if (!this.disabled) {
+					this.incrementDisabled = false
+				}
+			}
+		},
+		disabled(newVal) {
+			if (newVal) {
```

Comment:
i see why you did this but consider the following
if for some reason, disabled was externally set to `true` WHILE incrementDisabled and/or decrementDisabled were false (b/c min and max were not exceeded) 
then later disabled is set to false - which would make us skip this if-condition; the only way incrementDisabled would be then set to their correct values according to the value of min or max would be when currentValue changes (check scope of watcher on currentValue above)

So, have an else body in this watcher for this very rare scenario
OR ternaries: my point being the following
this.incrementDisabled = newVal?newVal:(newVal+this.step>this.max)
this.decremetnDisabled= newVal?newVal:(newVal-this.step<this.min)

having said this: the step prop is not reset to null with every increment/decrement action so this, too, is not right...

Reply:
i.e. to guard against using the step when no action is triggered now to change the value, make sure you reset it to null

Reply:
disabled() { this.forceUpdate() }?